packages/cosmic-swingset/test/unitTests/test-rendezvous.js

```diff
@@ -0,0 +1,85 @@
+// @ts-check
+import '@agoric/install-ses';
+import test from 'ava';
+import { makeRendezvousNamespace } from '../../lib/ag-solo/vats/rendezvous';
+
+const rendezvousFailed = {
+  instanceOf: Error,
+  message: /^Rendezvous with .* not completed/,
+};
+
+test('rendezvous with self', async t => {
+  const makeRendezvous = makeRendezvousNamespace();
+  const self = makeRendezvous('self');
+  const rendezvous = self.startRendezvous(self.getLocalAddress(), 'initiator');
+  const resultP = rendezvous.getResult();
+  t.throws(() => self.completeRendezvous('other', 'foo'), rendezvousFailed);
+  t.is(self.completeRendezvous(['self'], 'first'), 'initiator');
+  t.throws(() => self.completeRendezvous('self', 'second'), rendezvousFailed);
+  t.is(await resultP, 'first');
+});
```

Comment:
Probably a dumb question, but what is the purpose of sending data both directions if the purpose is to share an object reference (presumably one way?)

Reply:
It's two-way.  The rendezvous exchanges an object with the matching peer in both of the two directions.

Reply:
Sorry, is it the same object such that the sender receives their own object back (no new information for the sender)?

Reply:
> Sorry, is it the same object such that the sender receives their own object back (no new information for the sender)?

No, it's the object from the other side.

Reply:
So there are two objects being sent? Or is one direction expected to just return a success message? Again, sorry for the dumb question but I'm still having trouble pinning down the basics of how this is meant to be used. The tests show two strings being exchanged, but when do we want to exchange two strings?

Reply:
> So there are two objects being sent? Or is one direction expected to just return a success message?

Two objects of any marshallable type (including the default, which is `undefined`).  No need to return success, since the rendezvous promise will not resolve on the initiator's side and will throw on the completer's side if it could not be completed.

> Again, sorry for the dumb question

Not dumb!  Fundamental. :smile:

> but I'm still having trouble pinning down the basics of how this is meant to be used. The tests show two strings being exchanged, but when do we want to exchange two strings?

Strings were the simple case to test the logic.  I should write a test for exchanging objects, too.